document/src/vespa/document/select/simpleparser.cpp

```diff
@@ -6,16 +6,25 @@
 
 namespace document::select::simple {
 
-size_t eatWhite(const char * s, size_t len)
-{
-    size_t pos(0);
-    for (;(pos < len) && isspace(s[pos]); pos++);
-    return pos;
+namespace {
+    size_t eatWhite(const char *s, size_t len) {
+        size_t pos(0);
+        for (; (pos < len) && isspace(s[pos]); pos++);
+        return pos;
+    }
+
+    bool icmp(char c, char l) {
+        return tolower(c) == l;
+    }
 }
 
-bool icmp(char c, char l)
-{
-    return tolower(c) == l;
+void
+Parser::setRemaining(vespalib::stringref s, size_t fromPos) {
+    if ((fromPos + 1) < s.size()) {
+        _remaining = s.substr(fromPos);
+    } else {
+        _remaining = "";
```

Comment:
According to https://en.cppreference.com/w/cpp/string/basic_string_view/substr throws

> [`std::out_of_range`](https://en.cppreference.com/w/cpp/error/out_of_range) if `pos > size()`.

I.e. it's well-defined when `fromPos <= s.size()`. `s.substr(s.size())` thus returns a `string_view` with a length of zero, which is analogous to `_remaining = ""` in the else-case.

Maybe this could be simplified to `_remaining = s.substr(std::min(fromPos, s.size()))`?

Reply:
Indeed, thanks.